src/cryptsetup/cryptsetup.c

```diff
@@ -365,7 +365,7 @@ static int get_password(const char *vol, const char *src, usec_t until, bool acc
                 escaped_name = maj_min;
                 maj_min = NULL;
         } else
-                escaped_name = cescape(name);
+                escaped_name = cescape(src);
```

Comment:
escaped_name is really a misnomer now, as in no case it contains the "name" string anymore. It should get a less misleading name...

Also, why do the major/minor stuff at all here? it sounds unnecessary, we can just use 'src' directly, unconditionally, that should be equally expressive, no? can you drop disk_major_minor() entirely? or is there any benefit to include the major/minor in the idea rather than the src device string as is?

Reply:
Agreed, ```escaped_name``` is confusing. I will rename. As for removal of ```disk_major_minor()```, well I am not sure. Honestly I'd keep it as it is for now because I don't know what may break if I change it. IOW, my intention here is to fix what is clearly broken rather than doing rework along the way. Also ask-password files are our public API so I'd not change unless there is a _strong_ reason.

Reply:
Well, the two device paths are ultimately equivalent. I am pretty sure simpler code is better code, and constant code clean-up and refactoring a necessity for healthy projects. Hence please, remove the disk_major_minor() stuff along with your rework, otherwise things are just confusing, and that helps noone...